test/test_type.jl

```diff
@@ -25,6 +25,48 @@ isdir(outdir) && rm(outdir, recursive = true)
         end
     end
 
+    @timed_testset "TreeMesh & SerialTree type consistence" begin
+        # Helper function to test if the compiler can infer the `TreeType`
+        # which forms the basis for the `TreeMesh`.
+        # When this test passes, functions such as 
+        # `total_volume(mesh)` should also be inferable, see
```

Comment:
Can't you just test `total_volume(mesh)` additionally?

Reply:
Yes :+1: